primitives/src/transaction.rs

```diff
@@ -15,6 +15,7 @@ use core::fmt;
 #[cfg(feature = "arbitrary")]
 use arbitrary::{Arbitrary, Unstructured};
 use hashes::sha256d;
+#[cfg(feature = "alloc")]
```

Comment:
What does adding the alloc feature here have to do with the unused import?

Reply:
hmm ok.  I'm not sure what flags were used to get the unused import message here.

Reply:
I think adding the feature gate is correct, but there are other cases where it should also be added and it might be better to deal with them all as a separate PR/issue, rather than adding a seemingly random one here.

Reply:
found it with `--no-default-features`

Reply:
weird, running this on the current master branch doesn't produce the error for me.

Reply:
s/error/warning

Reply:
cd into the `primitives` dir first - that's how i found it

Reply:
> cd into the primitives dir first - that's how i found it

Ah yes I see now.  Although just removing `str` fixed the warning without adding to the test module or adding a feature gate.

Reply:
> cd into the primitives dir first - that's how i found it

I see, I think there are two separate warning, the alloc gate fixed the warning in primitives and str fixed the warning in bitcoin.

Reply:
I don't think the test module needs to be updated still, but I guess it doesn't hurt.

Reply:
> I think adding the feature gate is correct, but there are other cases where it should also be added and it might be better to deal with them all as a separate PR/issue, rather than adding a seemingly random one here.

Yes I agree, that would save some confusion when reviewing.